core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -400,9 +400,15 @@ trait WorkflowComponent {
       }
 
     def listWorkflowRecsForSubmissionAndStatuses(submissionId: UUID,
+                                                 limit: Int,
                                                  statuses: WorkflowStatuses.WorkflowStatus*
     ): ReadAction[Seq[WorkflowRecord]] =
-      findWorkflowsBySubmissionId(submissionId).filter(_.status inSetBind (statuses.map(_.toString))).result
+      findWorkflowsBySubmissionId(submissionId).filter(_.status inSetBind (statuses.map(_.toString))).take(limit).result
```

Comment:
the `.take(limit)` here should result in only returning `${workflowsPerMonitorPass}` workflows to the submission monitor actor for any given pass.

Reply:
The drawback to adding a limit here is that if the first `limit` rows from the db are still running, but limit+1, limit+2, limit+3 are done … the monitor won't complete the ones that are done. This isn't great.

Reply:
You could alleviate this by `select … order by rand() limit 20`; in mysql, to get a random selection of 20 rows that match the where clause. I'm not sure how to do that in Slick.